cmd/aries-js-worker/main.go

```diff
@@ -230,3 +191,159 @@ func defaultHandlers() map[string]map[string]func(*command) *result {
+	if err != nil {
+		return newErrResult(c.ID, err.Error()), false
+	}
+
+	ctx, err := a.Context()
+	if err != nil {
+		return newErrResult(c.ID, err.Error()), false
+	}
+
+	var notifier webhook.Notifier
+	if cOpts.Notifier != "" {
+		notifier = &jsNotifier{topic: cOpts.Notifier}
+	}
+
+	commands, err := controller.GetCommandHandlers(ctx, controller.WithMessageHandler(msgHandler),
+		controller.WithDefaultLabel(cOpts.Label), controller.WithNotifier(notifier))
+	if err != nil {
+		return newErrResult(c.ID, err.Error()), false
+	}
+
+	// add command handlers
+	addCommandHandlers(commands, pkgMap)
+
+	// add stop aries handler
+	addStopAriesHandler(a, pkgMap)
+
+	return &result{
+		ID:      c.ID,
+		Payload: "aries started",
+	}, true
+}
+
+func addCommandHandlers(commands []cmdctrl.Handler, pkgMap map[string]map[string]func(*command) *result) {
+	for _, cmd := range commands {
+		fnMap, ok := pkgMap[cmd.Name()]
+		if !ok {
+			fnMap = make(map[string]func(*command) *result)
+		}
+
+		fnMap[cmd.Method()] = cmdExecToFn(cmd.Handle())
+		pkgMap[cmd.Name()] = fnMap
+	}
+}
+
+func addStopAriesHandler(a *aries.Aries, pkgMap map[string]map[string]func(*command) *result) {
+	fnMap := make(map[string]func(*command) *result)
+	fnMap["Stop"] = func(c *command) *result {
+		err := a.Close()
+		if err != nil {
+			return newErrResult(c.ID, err.Error())
+		}
+
+		// reset handlers when stopped
+		for k := range pkgMap {
+			delete(pkgMap, k)
+		}
+
+		return &result{
+			ID:      c.ID,
+			Payload: "aries stoppped",
+		}
+	}
+	pkgMap["aries"] = fnMap
+}
+
+func newErrResult(id, msg string) *result {
+	return &result{
+		ID:     id,
+		IsErr:  true,
+		ErrMsg: msg,
+	}
+}
+
+func startOpts(str string) (*ariesStartOpts, error) {
+	opts := ariesStartOpts{}
+
+	err := json.Unmarshal([]byte(str), &opts)
+	if err != nil {
+		return nil, err
+	}
+
+	return &opts, nil
+}
+
+func ariesOpts(opts *ariesStartOpts) ([]aries.Option, error) {
+	msgHandler := msghandler.NewRegistrar()
+
+	var options []aries.Option
+	options = append(options, aries.WithMessageServiceProvider(msgHandler))
+
+	if opts.TransportReturnRoute != "" {
+		options = append(options, aries.WithTransportReturnRoute(opts.TransportReturnRoute))
+	}
+
+	for _, transport := range opts.OutboundTransport {
+		switch transport {
+		case "http":
+			outbound, err := arieshttp.NewOutbound(arieshttp.WithOutboundHTTPClient(&http.Client{}))
+			if err != nil {
+				return nil, err
+			}
+
+			options = append(options, aries.WithOutboundTransports(outbound))
+		case "ws":
+			options = append(options, aries.WithOutboundTransports(ws.NewOutbound()))
+		default:
+			return nil, fmt.Errorf("unsupported transport : %s", transport)
+		}
+	}
+
+	return options, nil
+}
+
+// jsNotifier notifies incoming events once registered
+type jsNotifier struct {
+	topic string
+}
+
+// Notify is mock implementation of webhook notifier Notify()
+func (n *jsNotifier) Notify(topic string, message []byte) error {
```

Comment:
why is the topic argument not sent to Javascript?

Reply:
Notifier follows REST pattern, it is initialized with URLs instead of Notifier implementation. I added new option in this PR `WithNotifier()` to support commands. REST doesn't have selective topic subscription feature, and also DIDComm msg handling fails when there isn't any webhook URL available. We may need to revisit webhook logic.

I can pass array of topic names from Javascript to notify by topic(protocol)